allennlp/common/file_utils.py

```diff
@@ -158,17 +161,21 @@ def cached_path(
             # This is the path the file should be extracted to.
             # For example ~/.allennlp/cache/234234.21341 -> ~/.allennlp/cache/234234.21341-extracted
             extraction_path = file_path + "-extracted"
+            extraction_lock_path = file_path + ".lock"
```

Comment:
I think this means we use the same lock file for downloading and extracting. I don't think that's ideal.

Reply:
Right. I thought a little bit about this, and I think it's actually the right thing to do.

If we used a different lock file for extraction, it's possible for one process to be writing to an archive file while another process starts trying to extract it, resulting in an extraction error.

Using the same lock file avoids this situation.

Reply:
Then again... since we download the archive to a temp file first then just rename it, that situation might be impossible anyway. But I don't see any downside to using the same lock file.

Reply:
Then again again... I guess one downside to using the same lock file is this:

Say you have one process (process "A") extracting the cached archive, and another process (process "B") that just wants the path to the cached archive, but doesn't want it extracted.

If process A is extracting the archive while process B is just trying to get the path to the archive, process B will have to wait until process A is finished extracting.

Reply:
Ok I changed to a different lock file for extraction ✅